packages/math-input/src/store/pager-reducer.ts

```diff
@@ -0,0 +1,141 @@
+import VelocityTracker from "../components/velocity-tracker";
+import {KeyTypes} from "../consts";
+import KeyConfigs from "../data/key-configs";
+
+import {defaultKeypadType, keypadForType} from "./shared";
+
+import type {PagerState} from "./types";
+
+// We default to the right-most page. This is done so-as to enforce a
+// consistent orientation between the view pager layout and the flattened
+// layout, where our default page appears on the far right.
+const getDefaultPage = (numPages) => numPages - 1;
+
+const initialPagerState = {
+    animateToPosition: false,
+    currentPage: getDefaultPage(keypadForType[defaultKeypadType].numPages),
+    // The cumulative differential in the horizontal direction for the
+    // current swipe.
+    dx: 0,
+    numPages: keypadForType[defaultKeypadType].numPages,
+    pageWidthPx: 0,
+    velocityTracker: new VelocityTracker(),
+} as const;
+
+const pagerReducer = function (
+    state = initialPagerState,
+    action: {
+        type: string;
+    },
```

Comment:
**suggestion:** this could use a better type to avoid all of the `@ts-expect-errors` below.